### PR TITLE
Update job

### PR DESCRIPTION
**Pull request by kihuni**
## Description

Implemented an API endpoint to update job post details using FastAPI. The endpoint allows authenticated users to update existing job posts by providing updated job details. Added appropriate permission checks to ensure only the job owner or an admin can update the job post. The endpoint handles both success and error responses based on the validity of the job post ID and the user's permissions.

​

## Related Issue (Link to issue ticket)

[approved issue](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/63](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/158)

​

## Motivation and Context

This change is required to allow users to correct or update job details in case of errors or changes. It ensures that job postings are kept accurate and up-to-date, improving the overall user experience and data integrity.
​

## How Has This Been Tested?

1. Successful update with a valid job ID and authenticated user.
2. Update attempt with an invalid job ID, returning a 404 Not Found response.
3. Unauthorized update attempt by a user who is not the owner or an admin, returning a 403 Forbidden response.
4. Update attempt with an invalid request body, returning a 422 Unprocessable Entity response.
​
Testing was done using pytest with a TestClient to simulate API requests and validate responses. The test environment included a mock database setup to ensure isolation and reliability of tests.

## Screenshots (if appropriate - Postman, etc):

​

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by Shullyd7** [comment]
You have conflicts. Resolve them and push again

File: api/v1/models/job.py [hunk 1]
```diff
@@ -21,15 +21,21 @@
 
 
 class Job(BaseModel, Base):
+
+    __tablename__ = "jobs"
+
+    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
+
```
Comment (Dev-wonderful) [inline api/v1/models/job.py:28]: Where is the migration for this

File: api/v1/routes/job.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+job = APIRouter(
+    prefix="/api/v1/jobs",
+    tags=["todos"],
+)
+
+
+@job.post(
+    "",
+    status_code=status.HTTP_201_CREATED,
```
Comment (Dev-wonderful) [inline api/v1/routes/job.py:22]: Why is the route empty, `/` would be okay

File: api/v1/routes/job.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    db.add(new_job)
+    db.commit()
+    db.refresh(new_job)
+
+    return {
+        "status": status.HTTP_201_CREATED,
+        "message": "Job listing created successfully",
+        "data": new_job,
+    }
```
Comment (Dev-wonderful) [inline api/v1/routes/job.py:44]: check `JsonResponseDict` for the standard response class based on the specified convention, update this all round

File: api/v1/routes/job.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    db.add(db_job_post)
+    db.commit()
+    db.refresh(db_job_post)
+
+    return {
+        "message": "Job details updated successfully",
+        "data": db_job_post,
+        "status_code": status.HTTP_200_OK
+    }
```
Comment (Dev-wonderful) [inline api/v1/routes/job.py:79]: Why are you working on 2 endpoints

File: main.py [hunk 5]
```diff
@@ -36,6 +43,10 @@ async def lifespan(app: FastAPI):
 
 app.add_exception_handler(CustomException, custom_exception_handler) # Newsletter custom exception registration
 
+
+app.include_router(auth)
+app.include_router(job)
+# app.include_router(users, tags=["Users"])
```
Comment (Dev-wonderful) [inline main.py:49]: Not acceptable check how it is being done in the routes init file

File: tests/v1/test_job.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+def override_get_db():
+    db = TestingSessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+app.dependency_overrides[get_db] = override_get_db
```
Comment (Dev-wonderful) [inline tests/v1/test_job.py:28]: Mock the database

**Review by Dev-wonderful — CHANGES_REQUESTED** [review]

File: api/v1/routes/job.py (hunk 4 above)
Comment (kihuni) [reply api/v1/routes/job.py:79]: I only worked on one; the other is for the person working on the create job endpoint.ndpoint.

**Comment by kihuni** [comment]
Let me fix everything.

**Pull request closed without merging**